modules/images/regenerate-existing-images/helper.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Functions used by the background process API.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Creates a background job.
+ *
+ * Job refers to the term and queue refers to the `background_job` taxonomy.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $action Name of job action.
+ * @param array  $data Optional. Arbitrary data for the job. Default empty array.
+ * @return Perflab_Background_Job|WP_Error Job object if created successfully, else WP_Error.
+ */
+function perflab_create_background_job( $action, array $data = array() ) {
+	// Ensure that job action is string.
+	if ( ! is_string( $action ) ) {
+		return new WP_Error( 'job_action_invalid_type', __( 'Job name must be string.', 'performance-lab' ) );
+	}
+
+	/**
+	 * Create the unique term name dynamically.
+	 *
+	 * Save job action. sanitize_title will be used before saving action.
+	 * Note that it will allow alphanumeric characters with underscores.
+	 * For instance, 'custom_job_action' or 'my_custom_123_job'.
+	 */
+	$term_name   = sanitize_title( $action );
+	$term_name   = str_replace( '-', '_', $term_name );
+	$term_object = (object) array(
+		'taxonomy' => PERFLAB_BACKGROUND_JOB_TAXONOMY_SLUG,
+		'parent'   => null,
+	);
+	$slug        = wp_unique_term_slug( $term_name, $term_object );
+	$term_data   = wp_insert_term( $term_name, PERFLAB_BACKGROUND_JOB_TAXONOMY_SLUG, array( 'slug' => $slug ) );
```

Comment:
This looks like it could be simplified.

`wp_insert_term` runs both `sanitize_title` and `wp_unique_term_slug`.

We can keep the `sanitize_title` and `str_replace` to ensure all actions passed are formatted correctly (alpha-numeric with underscores).

However we don't need to use the `wp_unique_term_slug` call as the `wp_insert_term` will handle this.

I would suggest the following updates

```suggestion
	$term_name = str_replace( '-', '_', sanitize_title( $action ) );
	$term_data = wp_insert_term( $term_name, PERFLAB_BACKGROUND_JOB_TAXONOMY_SLUG ) );
```